public/app/features/org/state/actions.ts

```diff
@@ -17,6 +18,7 @@ export function updateOrganization(): ThunkResult<any> {
 
     await getBackendSrv().put('/api/org', { name: organization.name });
```

Comment:
nitpick: you might want to move out `getBackendSrv` as a dependency like we do here https://github.com/grafana/grafana/blob/5a58f2b4c51b38c2fc03c1c5acf30b977c66271b/public/app/features/variables/state/actions.ts#L465 
This will make it easier to test `updateOrganization` without using jest.mock that can be limiting in some cases.

Reply:
I have addressed this in my latest commit. 

I also went ahead and added `getBackendSrv` as a dependency to `loadOrganization` as well, since it seemed to be causing my test to fail. 

One potential concern I want to call out is that I pass the `dependencies` parameter from `updateOrganization` to `loadOrganization`, which I am not sure is correct. Removing this caused me to get a type error when running my test that said `get` was `undefined`. Adding the `dependencies` parameter and passing it in from `updateOrganization` resolved that issue. Is there a better solution?

Reply:
Aha, good catch didn't notice that chain. I think it's fine as it is the same dependency. So you could declare one type `type OrganizationDependencies = { getBackendSrv: typeof getBackendSrv };` and use that in both thunks.

Another alternative is to remove the `loadOrganization` call in `updateOrganization` and add that call after this line https://github.com/grafana/grafana/blob/eed313f3129d2e1bb321ce28c6e859e623045f8b/public/app/features/org/OrgDetailsPage.tsx#L30

 I think this is the last comment I've left, outstanding work @KrammerJake  🌟

Reply:
Thank you! 🙇 I appreciate the helpful feedback! 

I implemented your first suggestion and consolidated the organization dependency type. I liked your second suggestion of moving the `loadOrganization` dispatch outside of `updateOrganization` thunk that you mentioned above; however, I was worried I potentially might introduce a race condition since we weren't `await`ing the thunk dispatches in `onUpdateOrganization` and the `get` request might return before the `put` finishes? I was thinking if we moved the `loadOrganization` call, we would likely want to rewrite `onUpdateOrganization` like this:

```js
  onUpdateOrganization = async (orgName: string) => {
    this.props.setOrganizationName(orgName);
    await this.props.updateOrganization();
    await this.props.loadOrganization();
  };
```

Let me know if you'd like me to make that change.